sockets/v0/Socket.h

```diff
@@ -62,53 +63,64 @@ class Socket {
      * Start the process of resolving a domain name.
      * If the input is a text IP address, an event is queued immediately; otherwise, onDNS is
      * queued as soon as DNS is resolved.
-     * The socket must have been opened before resolve is called
+     * The socket must have been opened before resolve is called. If it hasn't been opened then
```

Comment:
Hmm. It does require a valid API, though. This is something we will need to address. You shouldn't need an open socket to execute a DNS query.